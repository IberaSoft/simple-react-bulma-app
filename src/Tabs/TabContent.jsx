import React, { Component } from 'react';

class TabContent extends Component {
    clickTab(index, event) {
        this.props.clickTab(index);
    }
    render() {
        let activeClass = this.props.activeId;
        let tabContent = this.props.tabData.map((item, index) => {
            return <div 
                        className={'tab-pane ' + (activeClass === index ? "is-active" : "")}
                        key={index}>
                            {item.text}
                    </div>
        });
        return (
            <div className="tab-content">{tabContent}</div>
        );
    }
}

export default TabContent;