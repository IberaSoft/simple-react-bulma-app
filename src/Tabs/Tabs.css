.tab-pane {
    display: none;
}
.tab-pane.is-active {
    display: block;
}