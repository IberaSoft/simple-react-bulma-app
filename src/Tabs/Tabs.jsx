
import React, { Component } from 'react';

class Tabs extends Component {
    state = {
        activeIndex: 0
    }

    headerClick = (index) => {
        if (index !== this.state.activeIndex) this.setState({ activeIndex: index })
    }

    headerClass = (index) => (this.state.activeIndex === index ) ? "is-active" : ""

    render() {
        const { tabsItems } = this.props
        const { activeIndex } = this.state
      
        return (
            <div>
                <div className="tabs">
                    <ul>
                    { 
                        tabsItems.map((tab, index) =>
                            <li key={tab.title + index}
                                onClick={() => this.headerClick(index)}
                                className={this.headerClass(index)}>
                                <a>{tab.title}</a>
                            </li>
                        )
                    }
                    </ul>
                </div>
                <div className="tab-content">
                    <div className="tab-pane is-active">
                        {tabsItems[activeIndex].content}
                    </div>
                </div>
            </div>
        );
    }
}

export default Tabs;