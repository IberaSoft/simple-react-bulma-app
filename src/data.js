const menuItems = [
  { title: 'Tabs' },
  { title: 'Accordion' },
  { title: 'My IP' },
  { title: 'Menu 4' },
];

const dataItems = [
  { title: 'Tab 1', content: 'CONTENT 1 Lorem ipsum dolor sit amet consectetur adipisicing elit. Tempore magni magnam exercitationem laborum temporibus neque officia fugit, perferendis voluptatum, impedit repudiandae obcaecati animi ipsam. Eos culpa iure minima eum ad.' },
  { title: 'Tab 2', content: 'CONTENT 2 Lorem ipsum dolor sit amet consectetur adipisicing elit. Tempore magni magnam exercitationem laborum temporibus neque officia fugit, perferendis voluptatum, impedit repudiandae obcaecati animi ipsam. Eos culpa iure minima eum ad.' },
  { title: 'Tab 3', content: 'CONTENT 3 Lorem ipsum dolor sit amet consectetur adipisicing elit. Tempore magni magnam exercitationem laborum temporibus neque officia fugit, perferendis voluptatum, impedit repudiandae obcaecati animi ipsam. Eos culpa iure minima eum ad.' },
  { title: 'Tab 4', content: 'CONTENT 4 Lorem ipsum dolor sit amet consectetur adipisicing elit. Tempore magni magnam exercitationem laborum temporibus neque officia fugit, perferendis voluptatum, impedit repudiandae obcaecati animi ipsam. Eos culpa iure minima eum ad.' }
];

export  { menuItems, dataItems };