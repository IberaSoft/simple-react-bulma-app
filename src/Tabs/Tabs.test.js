import React from 'react'
import Tabs from './Tabs'
import ShallowRenderer from 'react-test-renderer/shallow';
import { dataItems } from './data';

describe('Tabs component', () => {
    it('Should match the snapshot', () => {
        const renderer = new ShallowRenderer();
        const result = renderer.render(<Tabs tabsItems={dataItems} />);
        expect(result).toMatchSnapshot();
    });
})