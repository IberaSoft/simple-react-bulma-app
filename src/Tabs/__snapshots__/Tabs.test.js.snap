// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component Should match the snapshot 1`] = `
<div>
  <div
    className="tabs"
  >
    <ul>
      <li
        className="is-active"
        onClick={[Function]}
      >
        <a>
          Tab 1
        </a>
      </li>
      <li
        className=""
        onClick={[Function]}
      >
        <a>
          Tab 2
        </a>
      </li>
      <li
        className=""
        onClick={[Function]}
      >
        <a>
          Tab 3
        </a>
      </li>
      <li
        className=""
        onClick={[Function]}
      >
        <a>
          Tab 4
        </a>
      </li>
    </ul>
  </div>
  <div
    className="tab-content"
  >
    <div
      className="tab-pane is-active"
    >
      CONTENT 1 Lorem ipsum dolor sit amet consectetur adipisicing elit. Tempore magni magnam exercitationem laborum temporibus neque officia fugit, perferendis voluptatum, impedit repudiandae obcaecati animi ipsam. Eos culpa iure minima eum ad.
    </div>
  </div>
</div>
`;
